src/components/Appbar/utils.ts

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import color from 'color';
+import { StyleSheet } from 'react-native';
+import type { ColorValue, StyleProp, ViewStyle } from 'react-native';
+import AppbarContent from './AppbarContent';
+import AppbarAction from './AppbarAction';
+import AppbarBackAction from './AppbarBackAction';
+import overlay from '../../styles/overlay';
+import type { Theme } from '../../types';
+import { black, white } from '../../styles/themes/v2/colors';
+
+export type AppbarModes = 'small' | 'medium' | 'large' | 'center-aligned';
+
+export const getAppbarColor = (
+  theme: Theme,
+  elevation: number,
+  customBackground?: ColorValue,
+  elevated?: boolean
+) => {
+  const { isV3, dark: isDarkTheme, mode, colors } = theme;
+  const isAdaptiveMode = mode === 'adaptive';
+  let backgroundColor;
+  if (customBackground) {
+    backgroundColor = customBackground;
+  } else if (isV3) {
+    if (elevated) {
+      backgroundColor = color(colors.surface)
+        .mix(color(colors.primary), 0.08)
+        .rgb()
+        .string();
+    } else {
+      backgroundColor = colors.surface;
+    }
+  } else if (!isV3) {
+    if (isDarkTheme && isAdaptiveMode) {
+      backgroundColor = overlay(elevation, colors?.surface);
+    } else backgroundColor = colors.primary;
+  }
+
+  return backgroundColor;
```

Comment:
Since it always should return just a string value for backgroundColor, I highly suggest putting guard clauses here to avoid nested if/else statements
```suggestion
  if (customBackground) {
    return customBackground;
  }
  
  if(!isV3) {
    if (isDarkTheme && isAdaptiveMode) {
      return overlay(elevation, colors?.surface);
    }
    
    return colors.primary;
  }
  
   if (elevated) {
     return color(colors.surface)
       .mix(color(colors.primary), 0.08)
       .rgb()
       .string();
   }
    
   return colors.surface;
```

Reply:
Agree, that makes sense, I think that in the initial version I assumed `getAppbarColor` will return more colors.